src/scripts/services/dimStoreService.factory.js

```diff
@@ -639,6 +655,29 @@ function StoreService(
       return null;
     }
 
+    if (itemDef.classified) {
+      var language = dimSettingsService.language;
+      if (dimClassifiedData[itemDef.itemHash]) { // do we have declassification info for item?
+         // itemDef.icon = dimClassifiedData[itemDef.itemHash].icon;
+        itemDef.itemName = dimClassifiedData[itemDef.itemHash].i18n[language].itemName;
+        itemDef.itemDescription = dimClassifiedData[itemDef.itemHash].i18n[language].itemDescription;
+        itemDef.itemTypeName = dimClassifiedData[itemDef.itemHash].i18n[language].itemTypeName;
+        itemDef.bucketTypeHash = dimClassifiedData[itemDef.itemHash].bucketHash;
+        itemDef.tierType = dimClassifiedData[itemDef.itemHash].tierType;
+        itemDef.classType = dimClassifiedData[itemDef.itemHash].classType;
+        if (dimClassifiedData[itemDef.itemHash].primaryBaseStatHash) {
+          itemDef.primaryBaseStatHash = dimClassifiedData[itemDef.itemHash].primaryBaseStatHash;
+          itemDef.primaryStat = [];
+          itemDef.primaryStat.statHash = itemDef.primaryBaseStatHash;
+          itemDef.primaryStat.value = dimClassifiedData[itemDef.itemHash].stats[itemDef.primaryStat.statHash].value;
+          item.primaryStat = itemDef.primaryStat;
+        }
+        if (dimClassifiedData[itemDef.itemHash].stats) {
+          item.stats = dimClassifiedData[itemDef.itemHash].stats;
+        }
+      }
+    }
```

Comment:
It'd be nice to move this into a function and use some temporary variables to clean it up.